src/web/core/src/main/java/org/geoserver/web/data/store/panel/FileParamPanel.java

```diff
@@ -60,7 +67,23 @@ public FileParamPanel(final String id, final IModel<String> paramValue, final IM
         add(label);
 
         // the text field, with a decorator for validations
-        textField = new TextField<String>("paramValue", new FileModel(paramValue));
+        FileRootsFinder rootsFinder = new FileRootsFinder(false);
+        textField = new AutoCompleteTextField<String>("paramValue", new FileModel(paramValue)) {
+            @Override
+            protected Iterator<String> getChoices(String input) {
+                try {
+                    // do we need to filter files?
+                    FileFilter fileFilter = FileParamPanel.this.fileFilter != null ?
+                            FileParamPanel.this.fileFilter.getObject() : null;
+
+                    return rootsFinder.getMatches(input, fileFilter).iterator();
+                } catch(Exception e) {
+                    // this is a helper, don't let it break the UI at runtime but log errors instead
+                    LOGGER.log(Level.SEVERE, "Failed to provide autocomplete for path " + input, e);
```

Comment:
Not sure SEVERE is required here, you are not going to shut down the application... can we tone this down to FINE before merge? If not I can see the logs filling up ...

Reply:
+1

Reply:
I've merged in the meantime, will direct commit the change to INFO